### PR TITLE
[receiver/windowseventlog] Add Windows Event Log Receiver

### PR DESCRIPTION
**Pull request by armstrmi**
**Description:** <Describe what has changed.>

Added Windows Event Logging Functionality based on the [corresponding log-collection operator](https://github.com/open-telemetry/opentelemetry-log-collection/tree/main/operator/input/windows).

**Link to tracking Issue:** <Issue number if applicable>
(https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/9225)

**Testing:** <Describe what testing was performed and which tests were added.>
`windowslog_test.go` verifies that event logs can be read from the Windows API

**Documentation:** <Describe the documentation added.>

(See README)

### CODE REVIEW

**Comment by jpkrohling** [comment]
This is still pending a sponsor.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,7 +7,7 @@
 ### 🚩 Deprecations 🚩
 
 ### 🚀 New components 🚀
-
+- `windowslogreceiver` Added implementation of Windows Event Log Receiver (#9228)
```
Comment (djaglowski) [inline CHANGELOG.md:10]:
```suggestion
- `windowseventlogreceiver` Added implementation of Windows Event Log Receiver (#9228)
```

File: receiver/windowslogreceiver/README.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		"computer": "example computer",
+		"channel": "application",
+		"record_id": 1,
+		"level": "Information",
+		"message": "example message",
+		"task": "example task",
+		"opcode": "example opcode",
+		"keywords": ["example keyword"]
+	}
+}
```
Comment (djaglowski) [inline receiver/windowslogreceiver/README.md:74]: Indentation needs attention.

File: receiver/windowslogreceiver/README.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+Configuration:
+```yaml
+- type: windowseventlog
+  channel: application
+```
+
+Output entry sample:
+```json
+{
+  "body": {
```
Comment (djaglowski) [inline receiver/windowslogreceiver/README.md:54]: The JSON shown here is a little more meaningful to the reader if it shown in context with more of the log structure.

The receiver also parses time and severity, so at least show those.

File: go.mod [hunk 4]
```diff
@@ -789,6 +788,8 @@ replace github.com/open-telemetry/opentelemetry-collector-contrib/receiver/udplo
 
 replace github.com/open-telemetry/opentelemetry-collector-contrib/receiver/wavefrontreceiver => ./receiver/wavefrontreceiver
 
+replace github.com/open-telemetry/opentelemetry-collector-contrib/receiver/windowslogreceiver => ./receiver/windowslogreceiver
```
Comment (djaglowski) [inline go.mod:791]: The package name should match the receiver name, so `windowseventlogreceiver`

File: go.mod [hunk 5]
```diff
@@ -653,8 +654,6 @@ replace github.com/open-telemetry/opentelemetry-collector-contrib/pkg/batchperre
 
 replace github.com/open-telemetry/opentelemetry-collector-contrib/pkg/batchpersignal => ./pkg/batchpersignal
 
-replace github.com/open-telemetry/opentelemetry-collector-contrib/pkg/winperfcounters => ./pkg/winperfcounters
```
Comment (djaglowski) [inline go.mod:656]: Should be fixed with a rebase

File: receiver/windowslogreceiver/testdata/simple.log [hunk 6]
```diff
@@ -0,0 +1,3 @@
+2020-08-25 INFO Something routine
```
Comment (djaglowski) [inline receiver/windowslogreceiver/testdata/simple.log:1]: Is this file used?

File: receiver/windowslogreceiver/windowslog.go [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+// Type is the receiver type
+func (f ReceiverType) Type() config.Type {
+	return typeStr
+}
+
+// CreateDefaultConfig creates a config with type and version
+func (f ReceiverType) CreateDefaultConfig() config.Receiver {
+	return createDefaultConfig()
+}
+func createDefaultConfig() *WindowsLogConfig {
```
Comment (djaglowski) [inline receiver/windowslogreceiver/windowslog.go:50]:
```suggestion
```

File: receiver/windowslogreceiver/windowslog_test.go [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+	_, err := NewFactory().CreateLogsReceiver(
+		context.Background(),
+		componenttest.NewNopReceiverCreateSettings(),
+		cfg,
+		new(consumertest.LogsSink),
+	)
+	require.Error(t, err, "receiver creation should fail if given invalid input config")
+}
+
+func TestReadWindowsEventLogger(t *testing.T) {
```
Comment (djaglowski) [inline receiver/windowslogreceiver/windowslog_test.go:80]: Nice test!

**Review by djaglowski — COMMENTED** [review]
Since this receiver can only run on windows, please add a [dummy implementation](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/journaldreceiver/journald_nonlinux.go) for non-windows systems, to ensure users get a clear error message.

**Comment by armstrmi** [comment]
Realize lots more files changed with this previous commit, will fix that momentarily

**Review by djaglowski — COMMENTED** [review]
Code looks good, but I've recently learned that windows unit tests are not currently running by default. For now, can you add a dedicated step to the `build-and-test-windows.yml`, like https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/9334/files#diff-5ca69371a0b431061a0b1b145a164ac5797c6a8bd745fba6061e3afe213318fd?

**Comment by armstrmi** [comment]
Will be holding off on closing this PR due to the following reasons.

During integration tests, we noticed that certain fields of the Windows Event Log was not being populated by the receiver. Upon further inspection we notice that in our implemented code, to scrape some of the fields: `message`,`level`, `opcode`, etc. from the `xml` format, it would be under the header `RenderingInfo`. Unfortunately that `RenderingInfo` header only applies to events that have been collected by the [Windows Event Collector](https://docs.microsoft.com/en-us/windows/desktop/WEC/windows-event-collector), otherwise it simply does not exist in the `XML` of the log. 

The underlying log-collection code that collects windows event logs should really collect from any service not just logs generated by the `Windows Event Collector` service. We have identified some changes we can make to the underlying code to make it more flexible.

We will be submitting an issue against the opentelemtry-log-collection repo detailing a proposed fix and work on the implementation.

This PR will be blocked by the above changes to the log-collection code.

**Comment by djaglowski** [comment]
@armstrmi, #10187 switched all of the other log-collection operators over to using `pkg/stanza` instead. Will you please update this PR to do the same?

**Review by djaglowski — COMMENTED** [review]
Let's simplify the file organization. We should follow the typical naming convention for files differentiated by OS, which is to append `_windows.go`, `_others.go`, and correspondingly `_windows_test.go` and `others_test.go`.

I see that you've deduplicated some code by having `windowslog.go` for all OS's, but it's easier to understand the overall layout of the operator if these two sets of files are cleanly separated.

File: receiver/windowseventlogreceiver/windowslog_test.go [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+		return sink.LogRecordCount() == 1
+	}
+
+	// logs sometimes take a while to be written, so a substantial wait buffer is needed
+	require.Eventually(t, logsReceived, 2*time.Minute, 200*time.Millisecond)
+	results := sink.AllLogs()
+	require.Len(t, results, 1)
+
+	records := results[0].ResourceLogs().At(0).ScopeLogs().At(0).LogRecords()
+	require.Equal(t, records.Len(), 1)
```
Comment (djaglowski) [inline receiver/windowseventlogreceiver/windowslog_test.go:116]: nit: Note the order of parameters should be `expected` then `actual`

File: receiver/windowseventlogreceiver/windowslog_test.go [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+	require.NoError(t, err)
+
+	logsReceived := func() bool {
+		return sink.LogRecordCount() == 1
+	}
+
+	// logs sometimes take a while to be written, so a substantial wait buffer is needed
+	require.Eventually(t, logsReceived, 2*time.Minute, 200*time.Millisecond)
+	results := sink.AllLogs()
+	require.Len(t, results, 1)
```
Comment (djaglowski) [inline receiver/windowseventlogreceiver/windowslog_test.go:113]: nit: Note the order of parameters should be `expected` then `actual`

**Review by djaglowski — COMMENTED** [review]
Aside from the file naming/organization changes and a couple nits, this looks good.

**Review by djaglowski — COMMENTED** [review]
This LGTM, except for the noted version issue. 

v0.52.0 is being released today. If you will make sure to match the versions used in #10298, then it may be possible to merge this before the release. Otherwise, it'll just be a trivial update to v0.52.0 after the release.

**Comment by djaglowski** [comment]
@armstrmi, looks good except a check is failing:

```
go.mod/go.sum deps changes detected, please run "make gotidy" and commit the changes in this PR.
Error: Process completed with exit code 1.
```

**Review by djaglowski — APPROVED** [review]
LGTM. Thanks @armstrmi!

**Pull request merged**